src/lib/logstore/log_store.cpp

```diff
@@ -416,13 +419,13 @@ void HomeLogStore::flush_sync(logstore_seq_num_t upto_seq_num) {
 
         // Step 2: After marking this lsn, we again do a check, to avoid a race where completion checked for no lsn
         // and the lsn is stored in step 1 above.
-        if (!m_records.status(upto_seq_num).is_active) { return; }
+        if (m_records.status(upto_seq_num).is_completed) { return; }
 
         // Step 3: Force a flush (with least threshold)
         m_logdev->flush_if_needed(1);
 
         // Step 4: Wait for completion
-        m_sync_flush_cv.wait(lk, [this, upto_seq_num] { return !m_records.status(upto_seq_num).is_active; });
+        m_sync_flush_cv.wait_for(lk, std::chrono::milliseconds(10), [this, upto_seq_num] { return m_records.status(upto_seq_num).is_completed; });
```

Comment:
I am assuming it is temporary, because caller expect data at the end of it. Either we return an error or we assert? In any case, after new flush changes, we probably don't need this cv so at that point it is moot.

Reply:
yes this is temporary for solving the racing we had seen. It is not right , but generally ensure we dont stuck on this CV infinitely for concurrent issue.